src/services/ChallengeService.js

```diff
@@ -2235,6 +2209,115 @@ deleteChallenge.schema = {
   challengeId: Joi.id(),
 };
 
+async function advancePhase(currentUser, challengeId, data) {
+  if (currentUser && (currentUser.isMachine || hasAdminRole(currentUser))) {
+    const challenge = await challengeDomain.lookup(getLookupCriteria("id", challengeId));
+
+    if (!challenge) {
+      throw new errors.NotFoundError(`Challenge with id: ${challengeId} doesn't exist.`);
+    }
+    if (challenge.status !== constants.challengeStatuses.Active) {
+      throw new errors.BadRequestError(
+        `Challenge with id: ${challengeId} is not in Active status.`
+      );
+    }
+
+    const phaseAdvancerResult = await phaseAdvancer.advancePhase(
+      challenge.id,
+      challenge.phases,
+      data.operation,
+      data.phase
+    );
+
+    if (phaseAdvancerResult.success) {
+      const grpcMetadata = new GrpcMetadata();
+
+      grpcMetadata.set("handle", currentUser.handle);
+      grpcMetadata.set("userId", currentUser.userId);
+
+      await challengeDomain.update(
+        {
+          filterCriteria: getScanCriteria({ id: challengeId }),
+          updateInput: {
+            phaseUpdate: {
+              phases: phaseAdvancerResult.updatedPhases,
+            },
+          },
+        },
+        grpcMetadata
+      );
+
+      const updatedChallenge = await challengeDomain.lookup(getLookupCriteria("id", challengeId));
+      await indexChallengeAndPostToKafka(updatedChallenge);
```

Comment:
we can avoid raising event unless there is a change in challenge status or winners are set